src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -81,47 +96,54 @@ int main(void) {
 }
 EOF
 
-  generate_gcno_files coverage_srcs/a.h coverage_srcs/a.cc coverage_srcs/t.cc
-  generate_instrumented_binary ./coverage_srcs/test coverage_srcs/a.h \
-      coverage_srcs/a.cc coverage_srcs/t.cc
-  generate_gcda_file ./coverage_srcs/test
+  generate_and_execute_instrumented_binary coverage_srcs/test \
+      "$COVERAGE_DIR_VAR/coverage_srcs" \
+      coverage_srcs/a.h coverage_srcs/a.cc coverage_srcs/t.cc
 }
 
-# Reads the list of arguments provided by the caller (using $@) and uses them
-# to produco .gcno files using g++.
-function generate_gcno_files() {
-  # "-fprofile-arcs -ftest-coverage" tells the compiler to generate coverage
-  # information needed by gcov and include additional code in the object files
-  # for generating the profiling.
-  g++ -fprofile-arcs -ftest-coverage "$@" && return 0
-  fail "Couldn't produce .gcno files for $@"
-  return 1
-}
-
-# Reads the list of arguments provided by the caller (using $@) and uses them
-# to produce an instrumented binary using g++.
+# Generates and executes an instrumented binary:
+#
+# Step 1: Reads the list of arguments provided by the caller (using $@) and uses them
+# to produce an instrumented binary using g++. This step also generates
+# the notes (.gcno) files.
+#
+# Step 2: Executes the instrumented binary. This step also generates the
```

Comment:
nit: Remove the "Step" prefixes. The documentation should read as a summary not an explicit list of steps that is likely to get outdated.

Reply:
Add (.gcda) after "profile data files" since it's the first time you mention them.

Reply:
Done and done.